### PR TITLE
[examples] Migrate development-to-production example to use Pythonic IO manager

### PR DESCRIPTION
**Pull request by benpankow**
## Summary

Moves over the resources used in the development-to-production example

### CODE REVIEW

**Comment by benpankow** [comment]
Current dependencies on/for this PR:
* master
  * **PR #13607** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13607" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/13607?utm_source=stack-comment).

**Review by yuhan — COMMENTED** [review]
looks great. might need to do a pass on https://docs.dagster.io/guides/dagster/transitioning-data-pipelines-from-development-to-production to eliminate all old usage in this example, e.g. i think we have `io_manager_key` set to `snowflake_io_manager` somewhere.

**Review by yuhan — APPROVED** [review]

File: docs/content/guides/dagster/transitioning-data-pipelines-from-development-to-production.mdx [hunk 1]
```diff
@@ -207,33 +201,29 @@ We still have some problems with this setup:
 1. Developers need to remember to change `user` and `password` to their credentials and `schema` to their name when developing locally.
 2. Passwords are being stored in code.
 
-We can easily solve these problems because the Snowflake I/O manager accepts configuration from environment variables using the <PyObject object="StringSource" /> configuration type. This allows us to store configuration values as environment variables and point Dagster to those environment variables in the run configuration:
+We can easily solve these problems using <PyObject object="EnvVar"/>, which lets us source configuration for resources from environment variables. This allows us to store Snowflake configuration values as environment variables and point the IO manager to those environment variables:
```
Comment (jamiedemaria) [inline docs/content/guides/dagster/transitioning-data-pipelines-from-development-to-production.mdx:204]:
```suggestion
We can easily solve these problems using <PyObject object="EnvVar"/>, which lets us source configuration for resources from environment variables. This allows us to store Snowflake configuration values as environment variables and point the I/O manager to those environment variables:
```
nit: I/O manager

**Review by jamiedemaria — APPROVED** [review]
lgtm

**Pull request merged**